### PR TITLE
Feature/bundle js with webpack

### PR DESCRIPTION
**Pull request by bsutt123**
Included a basic webpack and bundle.js configuration to get started. Other small points

1) added node_modules to .gitignore and excluded from jekyll again.
2) bundle.js is not ignored because we will have to include that to get the website working on deploy. In an ideal world, GH pages runs `npm run build` before building the jekyll page but I don't know if we can make it that smart.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,5 +2,10 @@ _site
 .jekyll-metadata
 .sass-cache
 feed.xml
+<<<<<<< HEAD
```
Comment (treznick) [inline .gitignore]: whoopsie!

File: _includes/head.html [hunk 2]
```diff
@@ -22,9 +22,7 @@
   <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
   <link href='https://fonts.googleapis.com/css?family=Montserrat|Open+Sans:400,700,400italic,700italic' rel='stylesheet' type='text/css'>
 
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/webcomponentsjs/0.7.21/webcomponents-lite.min.js"></script>
-  <script src="{{ "/node_modules/jquery/dist/jquery.min.js" | prepend: site.baseurl }}"></script>
-  <script src="{{ "/node_modules/mustache/mustache.min.js" | prepend: site.baseurl }}"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/webcomponentsjs/0.7.21/webcomponents-lite.min.js"></script><  <script src="/js/bundle.js" type="text/javascript"></script>
```
Comment (treznick) [inline _includes/head.html]: you've got a stray `<` in there

File: js/components/meetup-utils.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+
+
```
Comment (treznick) [inline js/components/meetup-utils.js]: probably no need for this whitespace

File: .gitignore [hunk 4]
```diff
@@ -3,8 +3,11 @@ _site
 .sass-cache
 feed.xml
 <<<<<<< HEAD
+<<<<<<< HEAD
 =======
 js/bundle.js
+=======
+>>>>>>> fixed small issues with webpack config
```
Comment (treznick) [inline .gitignore]: more merge conflict detritis

File: _includes/head.html [hunk 5]
```diff
@@ -22,7 +22,7 @@
   <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
   <link href='https://fonts.googleapis.com/css?family=Montserrat|Open+Sans:400,700,400italic,700italic' rel='stylesheet' type='text/css'>
 
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/webcomponentsjs/0.7.21/webcomponents-lite.min.js"></script><  <script src="/js/bundle.js" type="text/javascript"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/webcomponentsjs/0.7.21/webcomponents-lite.min.js"></script>
```
Comment (treznick) [inline _includes/head.html]: I'd fix this up into the prior commit

File: webpack/entry.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+console.log("hello world");
```
Comment (treznick) [inline webpack/entry.js]: probably doesn't need to be here

File: webpack/entry.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-console.log("hello world");
```
Comment (treznick) [inline webpack/entry.js]: I'd also fix this up into the previous commit

File: package.json [hunk 8]
```diff
@@ -4,13 +4,13 @@
...
+     "Dan Bernier <danbernier@gmail.com>",
      "Joel Nimety <jnimety@gmail.com>",
-+    "Josh Santos <josh@omnidapps.com>",
-+    "maxx1128 <maxx1128@gmail.com>",
-     "Zach Morek" <zmorek@gmail.com>"
-+  ],
+     "Josh Santos <josh@omnidapps.com>",
+    "maxx1128 <maxx1128@gmail.com>",
+     "Zach Morek <zmorek@gmail.com>"
+  ],
```
Comment (treznick) [inline package.json]: I fixed a lot of this on master. Maybe do a rebase?

File: .gitignore [hunk 9]
```diff
@@ -2,13 +2,6 @@ _site
...
 .sass-cache
 feed.xml
-<<<<<<< HEAD
-<<<<<<< HEAD
-=======
-js/bundle.js
-=======
->>>>>>> fixed small issues with webpack config
 node_modules
->>>>>>> initial webpack commit :
```
Comment (treznick) [inline .gitignore]: yeah I'd fix these changes up into the commits that introduced them :)

File: README.md [hunk 10]
```diff
@@ -18,7 +18,9 @@ $ yarn
 
 ## Run Application Server
 
+make sure to build the webpacks bundle.js before serving
 ```
+$ npm run build
```
Comment (treznick) [inline README.md]: This maybe belongs in its own commit?

File: _includes/footer.html [hunk 11]
```diff
@@ -18,14 +18,10 @@ <h4 class="footer-heading">
           </h4>
         </a>
       </div>
-
     </div>
-
```
Comment (treznick) [inline _includes/footer.html]: As do these?

File: _includes/footer.html [hunk 12]
```diff
@@ -18,14 +18,10 @@ <h4 class="footer-heading">
...
   </div>
 
-  <script src="{{ "/js/modernizr.min.js" | prepend: site.baseurl }}"></script>
-
-  <script type="text/javascript" src="{{ '/js/bundle.js' | prepend: site.baseurl }}" charset="utf8" ></script>
+   <script src="{{ "/js/modernizr.min.js" | prepend: site.baseurl }}"></script>
```
Comment (treznick) [inline _includes/footer.html]: I'd fix these whitespace changes up into the commits that introduce them

File: webpack.config.js [hunk 13]
```diff
@@ -19,11 +19,10 @@ module.exports = {
...
     ]
   },
-  plugins: [
+  plugins:[
     new webpack.ProvidePlugin({
-      $: "jquery",
-      jQuery: "jquery",
-      "window.jQuery": "jquery"
+      $: 'jquery',
+      jQuery: 'jquery'
```
Comment (treznick) [inline webpack.config.js]: fixup into commit that introduce them

File: README.md (hunk 10 above)
Comment (NerdsvilleCEO) [reply README.md]: Yeah I think also we can clean up the commit history here with an interactive rebase, and split them up logically

File: package.json (hunk 8 above)
Comment (NerdsvilleCEO) [reply package.json]: +1 on this, also, for anything that isn't fixed, we should make sure the whitespace is consistent

File: _includes/footer.html [hunk 14]
```diff
@@ -18,18 +18,17 @@ <h4 class="footer-heading">
...
         </a>
       </div>
-
     </div>
-
   </div>
 
-  <script src="{{ "/node_modules/bootstrap-sass/assets/javascripts/bootstrap.min.js" | prepend: site.baseurl }}"></script>
-  <script src="{{ "/js/modernizr.min.js" | prepend: site.baseurl }}"></script>
+   <script src="{{ "/js/modernizr.min.js" | prepend: site.baseurl }}"></script>
```
Comment (NerdsvilleCEO) [inline _includes/footer.html]: Let's fix this spacing

**Comment by NerdsvilleCEO** [comment]
@bsutt123, either myself or @treznick are available to pair on this to get it ready to merge, all good changes so I personally would like to see this in the repo after some cleanup

**Comment by bsutt123** [comment]
I've been a bit busy with outside projects this week. So it looks like most of the requested changes are whitespace and small typing errors. Is it better to go back and revert to commits that only include the right whitespace, or make a new commit on top of what I have that fixes the whitespace but doesn't remove the original commits from the history?

Just curious which was is best.

**Comment by NerdsvilleCEO** [comment]
@bsutt123 it is commit splitting, commit message changes, and whitespace yeah, everything else basically seems good afaict

**Comment by bsutt123** [comment]
honestly, probably best to just figure out a time to pair on it, I haven't done that much before and I would probably just insert more problems.

**Comment by NerdsvilleCEO** [comment]
When is good for you I can set aside time later this week

**Comment by bsutt123** [comment]
Anytime after 8 works for me. Or earlier on some days if you don't stay up till 8.

**Comment by NerdsvilleCEO** [comment]
@treznick This looks much better now I worked with @bsutt123 to clean stuff up

File: README.md (hunk 10 above)
Comment (treznick) [reply README.md]: I think here we need to be explicit about what needs to happen before we release changes. This adds a step to our release workflow where we need to run `npm build` and then commit the compiled assets in a release branch. Let's chat in the slack about this.

**Comment by bsutt123** [comment]
Build now includes @NerdsvilleCEO work to use Travis CI to build. Merged into my branch on [my branch](https://github.com/bsutt123/newhavenio.github.io/pull/1)

**Comment by bsutt123** [comment]
So whats going on with this? Is there something we need to add or change?

**Comment by bsutt123** [comment]
So @NerdsvilleCEO added in TravisCI so that there isn't any pre-deploy instructions, the github pages deploy should take care of it. Josh added that as the later part of the commit because people were worried about the bundle.js needing to get built before deploying.

I added in a part to the readme that reminds you to do a build step locally if you want to make changes to `index.js` which is the entry point for the webpack build.

**Comment by jnimety** [comment]
oh, right! This might be ready then, would you mind looking it over to double check things?

**Comment by bsutt123** [comment]
I'll pull it down and run through it one more time just to be sure that everything is tip top shape.

**Comment by bsutt123** [comment]
switched to `feature/bundle-js-with-webpack` on my local computer and ran the following commands...

```
bundle install
yarn 
npm run build
jekyll serve
```
No errors, and the site was fully funcitonal. I'm not sure how to tell it to follow the TravisCI protocol directly, or I would have done that.

**Comment by jnimety** [comment]
@tom @NerdsvilleCEO do we have to do anything to enable the travic-ci deploy once we merge?

**Comment by treznick** [comment]
@jnimety I think it's probably good to go. I say we try it and if it blows up we can revert the build.

**Comment by bsutt123** [comment]
Is everyone ready to pull the trigger?

Also side note, the prodcution site can't currently find `slack-button.html` so that bug is coming from master. I'll see If i can track it down.

**Comment by ZachBeta** [comment]
@bsutt123 please smoke test and confirm

**Comment by ZachBeta** [comment]
the event list at the bottom seems to be failing, but idk if that is on my end, or was from this PR, or is intermittent, or what
